File: app/schemas/unit.py
"""Unit-of-measurement schemas."""
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class UnitBase(BaseModel):
    name: str = Field(min_length=1, max_length=50)
    short_name: str = Field(min_length=1, max_length=20)
    conversion_factor: Decimal | None = Field(default=None, ge=0)
    is_active: bool = True


class UnitCreate(UnitBase):
    pass


class UnitUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=50)
    short_name: str | None = Field(default=None, min_length=1, max_length=20)
    conversion_factor: Decimal | None = Field(default=None, ge=0)
    is_active: bool | None = None


class UnitOut(UnitBase):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: frontend/src/types/unit.ts
export interface Unit {
  id: number;
  name: string;
  short_name: string;
  conversion_factor: string | null;
  is_active: boolean;
}

export interface UnitFormValues {
  name: string;
  short_name: string;
  conversion_factor?: number;
  is_active: boolean;
}
